pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -258,6 +270,8 @@ func (rc *reconciler) attachDesiredVolumes() {
 				klog.Infof(volumeToAttach.GenerateMsgDetailed("Volume attached--touching", ""))
 			}
 			rc.actualStateOfWorld.ResetDetachRequestTime(volumeToAttach.VolumeName, volumeToAttach.NodeName)
+			// remove any pre-existed cache entry in this case. The volume has been attached
+			rc.operationStartTimeCache.Delete(createOperationUniqueKey(volumeToAttach.VolumeName, volumeToAttach.NodeName))
```

Comment:
It should not however there is a bug in the implementation I just fixed based on this.
The actualStateOfWorld will only be updated after the metric has been reported in operation_generator.go
I am adding the deletion here just to clean up the cache to avoid potential dead entry when the following situation happens:
1. controller error'd out trying to attach the volume to the node
2. volume has been attached to the node out of the bound
3. controller found the volume has been attached and stopped calling Attach
IMHO, the metric should not be reported as it may introduce unnecessary noises, however the cache should have been cleaned.